cmux-tui/scripts/iroh-docker-demo/attach-once.py

```
@@ -0,0 +1,229 @@
+            "id": request_id,
+            "method": method,
+            "params": params,
+        }
+        assert self.process.stdin is not None and self.process.stdout is not None
+        self.process.stdin.write(json.dumps(envelope) + "\n")
+        self.process.stdin.flush()
+        deadline = time.monotonic() + timeout
+        while time.monotonic() < deadline:
+            line = self.process.stdout.readline()
+            if not line:
+                raise RuntimeError(f"control process closed during {method}")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)